std/assembly/finalization.ts

```diff
@@ -0,0 +1,83 @@
+/**
+ * This method is a global function that works like a finalization
+ * callback when a reference is cleaned up.
+ *
+ * @param {usize} ptr - the pointer being finalized.
+ */
+
+// @ts-ignore: global
+@global @lazy
+export function __finalize(ptr: usize): void {
+
+    // then check to see if a reference is located on a registry
+    if (finalizationRegistryMap.has(ptr)) {
+        // get the registry associated
+        let registry = finalizationRegistryMap.get(ptr);
+
+        // virtual call and handle the pointer
+        registry.handle(ptr);
+
+        // remove it from the registry
+        finalizationRegistryMap.delete(ptr);
+    }
```

Comment:
Thanks for the help with code style problems. I'm trying my best here to remain on target with the "quality" of the code rather than the look of it, only because I want to make sure this all gets done correctly.